### PR TITLE
Set service account correctly in oadm registry, deprecate --credentials

### LINKED ISSUE #8285: `oadm registry --credentials` drops service account name
I have followed instructions from here - 

https://docs.openshift.org/latest/install_config/install/advanced_install.html#running-the-advanced-installation for openshift installation where I have 2 nodes and master is one of those 2 nodes.

I am attempting to start a docker registry using command -

sudo oadm registry --service-account=registry --config=/etc/origin/master/admin.kubeconfig --credentials=/etc/origin/master/openshift-registry.kubeconfig --mount-host=/var/openshift-docker-registry-storage

/var/openshift-docker-registry-storage has 777 permissions and I have also tried making 1001 owner of it.

I have also disabled SELinux completely to rule if it's playing up.

"registry" service account is allowed to run privileged containers by executing command 
oadm policy add-scc-to-user privileged registry and this was done before oadm registry .... command was executed.

Docker registry always failed to start with error that's described here -
https://bugzilla.redhat.com/show_bug.cgi?id=1238954

##### Version
oc v1.1.4
kubernetes v1.2.0-origin-41-g91d3e75

##### Steps To Reproduce
1. Creating docker registry by following steps from here using --mount-host https://docs.openshift.org/latest/install_config/install/docker_registry.html#deploy-registry

##### Current Result
registry POD times out stating unable to scale and timedout on condition 

##### Expected Result
registry POD should startup successfully.

### PR DESCRIPTION
**Pull request by liggitt**
Fixes #8285

### CODE REVIEW

**Comment by liggitt** [comment]
[test]
@smarterclayton PTAL

**Comment by smarterclayton** [comment]
@aweiteka please take a look at the atomic-quickstart changes

File: images/atomic-registry-quickstart/run.sh [hunk 1]
```diff
@@ -26,7 +26,7 @@ echo "Starting registry services..."
 
 set -x
 
-$CMD oadm registry --credentials /etc/origin/master/openshift-registry.kubeconfig
+$CMD oadm registry
```
Comment (aweiteka) [inline images/atomic-registry-quickstart/run.sh]: Per pending PR #8245 this dir was moved to examples/atomic-registry-quickstart

Comment (liggitt) [inline images/atomic-registry-quickstart/run.sh]: @aweiteka can you make the atomic example changes in your PR, and I'll remove them from this one, then?

Comment (aweiteka) [inline images/atomic-registry-quickstart/run.sh]: Done.

**Comment by liggitt** [comment]
removed atomic changes, updated `install_router` to not pre-create the service account which `oadm router` will now attempt to create

**Comment by smarterclayton** [comment]
LGTM

**Comment by liggitt** [comment]
[merge]

**Comment by kargakis** [comment]
This also needs a doc PR

**Pull request merged**